### PR TITLE
Fix autologging of Keras optimizers for TF.Keras 1.X

### PR DESCRIPTION
**Pull request by juntai-zheng**
## What changes are proposed in this pull request?

Originally, MLflow would only autolog TensorFlow optimizers when using Tensorflow.Keras models. This change allows for the autologging of Keras optimizers as well.

## How is this patch tested?

Changed the test in `tests/tensorflow_autolog/test_tensorflow_autolog.py` to reflect this change.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Keras optimizers are now correctly logged when autologging with TensorFlow.Keras 1.X models.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tensorflow.py [hunk 1]
```diff
@@ -471,13 +471,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
         opt = self.model.optimizer
-        if hasattr(opt, 'optimizer'):
+        # Secondary checks are if the optimizer is a TensorFlow optimizer rather than a Keras one.
```
Comment (smurching) [inline mlflow/tensorflow.py:474]: Can we pull the reassignment of `opt` up into its own if-statement before logging any params & explain why we do it? e.g.

```
# Explain why we do this
if hasattr(opt, 'optimizer'):
    opt = opt.optimizer
# Log optimizer attributes...
if hasattr(opt, '_name'):
```

Or if there's a reason we bundle assigning `opt` & logging the `name` together in the same branch, could we add a clarifying comment (e.g. because we want to log the name of the outer optimizer, but extract the inner optimizer so that we can log its learning rate & epislon)? Or maybe just log the name first, and then reassign `opt` afterwards? I think either would help make this clearer

**Review by smurching — COMMENTED** [review]
One comment, otherwise looks good

**Review by smurching — APPROVED** [review]
LGTM
